### PR TITLE
Row order fix

### PR DESCRIPTION
**Pull request by zander-prinsloo**
This PR fixes two small bugs in the `joyn::joyn()`.

1. The input keys in input tables were being renamed to "keyby1" if the input tables did not have the same row names. [See this task](https://app.clickup.com/t/8687ehx5f). 
2. The order of returned data frames were not correct. See [this task](https://app.clickup.com/t/86879u7kk).

### CODE REVIEW

**Comment by zander-prinsloo** [comment]
image

File: R/joyn-merge.R [hunk 1]
```diff
@@ -212,12 +212,8 @@ joyn <- function(x,
...
   start_joyn <- Sys.time()
   # copy objects if data.tables
-  if (any(class(x) == "data.table")) {
-    x <- copy(x)
-  }
-  if (any(class(y) == "data.table")) {
-    y <- copy(y)
-  }
+  x <- copy(x)
+  y <- copy(y)
```
Comment (randrescastaneda) [inline R/joyn-merge.R:216]: Why did you remove the condition? If the data is big, this `copy()` will slow down the function. You can use `inherits(x, "data.table")`.

**Review by randrescastaneda — APPROVED** [review]
everything looks great. I just added a small comment because I don't understand why the condition was removed. besides that, all great. Thanks

File: R/joyn-merge.R (hunk 1 above)
Comment (zander-prinsloo) [reply R/joyn-merge.R:216]: When using check_by_vars() it was modifying the names of the input tables by reference, so that the column names in x and y had been changed to `keyby1`. Which is why I used `copy()` internally. I can look for an alternative solution that is more memory efficient for large data frames

Comment (randrescastaneda) [reply R/joyn-merge.R:216]: GOt it. I think we can proceed for now, but that `data.table` code should be avoided. Thanks.
